pywr/hashes.py

```diff
@@ -20,17 +20,27 @@ class HashMismatchError(IOError):
     pass
 
 
-def check_hash(filename, hash, algorithm="md5", **kwargs):
+def check_hash(model, filename, hash, algorithm="md5", **kwargs):
     """Check the hash for filename using the named algorithm
 
     If the hashes do not match a HashMismatchError error is raised.
 
     This function is not case sensitive.
     """
 
+    if (cached_hash := model._file_hashes.get((filename, algorithm))) is not None:
```

Comment:
What's your opinion on this being a method on `Model`? It is bad practice to access private attributes of an object.

Reply:
yes, it probably is. I've moved it. Do you think its worth keeping this as a standalone module or could the rest of the code be moved as well?